src/ext/memory_limit.c

```diff
@@ -51,3 +51,16 @@ int64_t ddtrace_get_memory_limit(TSRMLS_D) {
 
     return limit;
 }
+
+BOOL_T ddtrace_check_memory_under_limit(TSRMLS_D) {
+    static int64_t limit = -1;
+    static zend_bool fetched_limit = 0;
+    if (!fetched_limit) {  // cache get_memory_limit() result to make this function blazing fast
```

Comment:
This isn't threadsafe, correct?

Reply:
Not explicitly. This is code that was [moved from **ddtrace.c**](https://github.com/DataDog/dd-trace-php/blob/master/src/ext/ddtrace.c#L480-L496).